Mempool requirement removed from transactions api regestration.

philip-morlier opened the pull request.

philip-morlier committed:
Mempool requirement removed from transactions api regestration.

philip-morlier committed:
Added comment in the case of no mempool

AusIV commented on main.go:
```diff
@@ -283,7 +285,7 @@ func main() {
 		tm.Register("eth", api.NewGasAPI(logsdb, cfg.Chainid, pl, cfg))
 	}
-	if hasTx && hasBlocks && hasLogs && hasMempool {
+	if hasTx && hasBlocks && hasLogs {
```
Do tests pass with this? I'm suspicious that some of the API calls will generate errors because they expect to be able to query the mempool table and it doesn't exist.

philip-morlier committed:
Changes made to enable api operation without mempool database.

A boolean was added to the transactions, flume, and gas api objects to allow the apis to modify their behavior based on the presence of a mempool.
Changes were also made to the api and polygon tests to correspond the the above.

philip-morlier committed:
Moved the evaluation of mempool boolean into utility functions away from top level api functions.
